### PR TITLE
[Model]: Support Helios  from ByteDance

### PR DESCRIPTION
**Pull request by princepride**
## Purpose

Add full support for the [Helios](https://github.com/PKU-YuanGroup/Helios.git) video generation model family in vLLM-Omni, covering **3 model variants** (Helios-Base, Helios-Mid, Helios-Distilled) across **3 generation tasks** (T2V, I2V, V2V) — a complete 3×3 support matrix.

| Variant | Pipeline Class | Key Features |
|---------|---------------|--------------|
| Helios-Base | `HeliosPipeline` | Stage 1 single-stage denoising, `guidance_scale=5.0` |
| Helios-Mid | `HeliosPyramidPipeline` | Stage 2 pyramid denoising, CFG-Zero* |
| Helios-Distilled | `HeliosPyramidPipeline` | Stage 2 pyramid + DMD few-step inference, `guidance_scale=1.0` |

## Test Plan

Manual end-to-end testing across all supported model×task combinations, NPU is supported as well. Run from the repository root:

**T2V (Text-to-Video):**
```bash
# Helios-Base (Stage 1)
python3 examples/offline_inference/helios/end2end.py \
--sample-type t2v \
--model BestWishYsh/Helios-Base \
--prompt "A vibrant tropical fish swimming gracefully among colorful coral reefs in a clear, turquoise ocean. The fish has bright blue and yellow scales with a small, distinctive orange spot on its side, its fins moving fluidly. The coral reefs are alive with a variety of marine life, including small schools of colorful fish and sea turtles gliding by. The water is crystal clear, allowing for a view of the sandy ocean floor below. The reef itself is adorned with a mix of hard and soft corals in shades of red, orange, and green. The photo captures the fish from a slightly elevated angle, emphasizing its lively movements and the vivid colors of its surroundings. A close-up shot with dynamic movement." \
--num-frames 600 \
--seed 42 \
--output helios_t2v_base.mp4

# Helios-Mid (Stage 2 + CFG-Zero*)
python examples/offline_inference/helios/end2end.py \
    --model BestWishYsh/Helios-Base --sample-type t2v \
    --prompt "A dynamic time-lapse video showing the rapidly moving scenery from the window of a speeding train." \
    --guidance-scale 5.0 --is-enable-stage2 \
    --pyramid-num-inference-steps-list 20 20 20 \
    --use-cfg-zero-star --use-zero-init --zero-steps 1 \
    --output helios_t2v_mid.mp4

# Helios-Distilled (Stage 2 + DMD)
python examples/offline_inference/helios/end2end.py \
    --model BestWishYsh/Helios-Base --sample-type t2v \
    --prompt "A dynamic time-lapse video showing the rapidly moving scenery from the window of a speeding train." \
    --num-frames 240 --guidance-scale 1.0 --is-enable-stage2 \
    --pyramid-num-inference-steps-list 2 2 2 \
    --is-amplify-first-chunk --output helios_t2v_distilled.mp4
```

**I2V (Image-to-Video):**
```bash
# Helios-Base
python examples/offline_inference/helios/end2end.py \
    --model ./Helios-Base --sample-type i2v \
    --image-path wave.jpg \
    --prompt "A towering emerald wave surges forward, its crest curling with raw power and energy." \
    --guidance-scale 5.0 --output helios_i2v_base.mp4
```

**V2V (Video-to-Video):**
```bash
# Helios-Base
python examples/offline_inference/helios/end2end.py \
    --model ./Helios-Base --sample-type v2v \
    --video-path car.mp4 \
    --prompt "A bright yellow Lamborghini speeds along a curving mountain road." \
    --guidance-scale 5.0 --output helios_v2v_base.mp4
```

## Test Result
**Helios-Base, T2V**

<details>

https://github.com/user-attachments/assets/8bb65d99-8669-46cb-ba15-c2802cb73eb3

</details>

**Helios-Mid, T2V**

<details>

https://github.com/user-attachments/assets/fc1af8bd-68b4-4d95-bb73-6052369d43f7

</details>

**Helios-Distilled, T2V**

<details>

https://github.com/user-attachments/assets/d6b40aeb-af6d-4edf-92eb-bd0a35787d77

</details>

**Helios-Base, I2V**

<details>

https://github.com/user-attachments/assets/caebd2ea-3577-4f06-b246-aac6ab27b4e2

</details>

**Helios-Base, V2V**

<details>

https://github.com/user-attachments/assets/a29c25fa-e8e4-49fa-ac34-9578558fa8c4

</details>

### CODE REVIEW

File: vllm_omni/diffusion/models/helios/pipeline_helios.py [hunk 1]
```diff
@@ -0,0 +1,1062 @@
...
+                    + (1 - frame_sigmas) * latent_chunk
+                )
+                noisy_latents_chunks.append(noisy_chunk)
+            video_latents = torch.cat(noisy_latents_chunks, dim=2)
+
+        # Prepare latent variables
+        num_channels_latents = self.transformer.config.in_channels
+        window_num_frames = (num_latent_frames_per_chunk - 1) * self.vae_scale_factor_temporal + 1
+        num_latent_chunk = max(1, (num_frames + window_num_frames - 1) // window_num_frames)
+        num_history_latent_frames = sum(history_sizes)
```
[the remark this answers is not in the document]
Comment (princepride) [reply vllm_omni/diffusion/models/helios/pipeline_helios.py:449]: These code copied from original code.

**Comment by hsliuustc0106** [comment]
## vllm-omni-reviewer Code Review

Found 3 critical issues:

### 1. Documentation formatting bug breaks table rendering

Line 10 has an orphaned `HeliosPyramidPipeline` outside the markdown table that breaks rendering. This appears to be a copy-paste error - the pipeline name is already in the table row above.

https://github.com/vllm-project/vllm-omni/blob/83ce0872685aa373966665f8658b91f23d4a50c5/docs/models/supported_models.md#L9-L11

**Fix:** Delete the orphaned line.

### 2. Missing test coverage for new model

This PR adds 3362 lines of new model code (transformer, pipeline, scheduler) but includes no test files. Per conventions, new model implementations require correctness validation tests.

https://github.com/vllm-project/vllm-omni/blob/83ce0872685aa373966665f8658b91f23d4a50c5/vllm_omni/diffusion/models/helios/

**Fix:** Add tests validating model loading and basic inference for the Helios variants.

### 3. Memory/hardware requirements not documented

The PR adds 3 model variants (Base, Mid, Distilled) with different memory profiles but doesn't document minimum VRAM, recommended GPUs, or memory configuration settings. Users have no guidance on hardware requirements.

https://github.com/vllm-project/vllm-omni/blob/83ce0872685aa373966665f8658b91f23d4a50c5/docs/models/supported_models.md#L9

**Fix:** Add memory requirements documentation for each variant (e.g., "Helios-Base: X GB VRAM minimum").

---

🤖 Generated with [vllm-omni-reviewer](https://github.com/hsliuustc0106/vllm-omni-reviewer)

File: docs/models/supported_models.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ th {
 |`ZImagePipeline` | Z-Image | `Tongyi-MAI/Z-Image-Turbo` |
 | `WanPipeline` | Wan2.2-T2V, Wan2.2-TI2V | `Wan-AI/Wan2.2-T2V-A14B-Diffusers`, `Wan-AI/Wan2.2-TI2V-5B-Diffusers` |
 | `WanImageToVideoPipeline` | Wan2.2-I2V | `Wan-AI/Wan2.2-I2V-A14B-Diffusers` |
+| `HeliosPipeline`, ``HeliosPyramidPipeline`` | Helios | `BestWishYsh/Helios-Base`, `BestWishYsh/Helios-Mid`, `BestWishYsh/Helios-Distilled` |
```
Comment (hsliuustc0106) [inline docs/models/supported_models.md]: Missing memory/hardware requirements. The PR adds 3 model variants (Base, Mid, Distilled) with different memory profiles but doesn't document minimum VRAM or recommended GPUs. Add a column or note specifying requirements for each variant.

File: docs/models/supported_models.md [hunk 3]
```diff
@@ -30,6 +30,8 @@ th {
...
+HeliosPyramidPipeline
```
Comment (hsliuustc0106) [inline docs/models/supported_models.md]: This orphaned `HeliosPyramidPipeline` line breaks the markdown table rendering. It appears to be a copy-paste error - the pipeline name is already in the table row above. **Delete this line.**

File: docs/models/supported_models.md (hunk 2 above)
Comment (princepride) [reply docs/models/supported_models.md]: 3 model have same memory profiles

**Comment by Gaohan123** [comment]
The link https://github.com/BestWishYsh/Helios cannot open

**Comment by princepride** [comment]
The model still didn't open source, please wait

**Comment by gcanlin** [comment]
```
python3 examples/offline_inference/helios/end2end.py \
--sample-type t2v \
--model ./Helios-Base \
--prompt "A vibrant tropical fish swimming gracefully among colorful coral reefs in a clear, turquoise ocean. The fish has bright blue and yellow scales with a small, distinctive orange spot on its side, its fins moving fluidly. The coral reefs are alive with a variety of marine life, including small schools of colorful fish and sea turtles gliding by. The water is crystal clear, allowing for a view of the sandy ocean floor below. The reef itself is adorned with a mix of hard and soft corals in shades of red, orange, and green. The photo captures the fish from a slightly elevated angle, emphasizing its lively movements and the vivid colors of its surroundings. A close-up shot with dynamic movement." \
--num-frames 600 \
--seed 42 \
--output helios_t2v_base.mp4 \
--tensor-parallel-size 2
```

NPU needs to enable TP=2. And end-to-end can run successfully.

**Comment by Sy0307** [comment]
If Helios has any plans for optimization on vllm-omni in the future, I would be happy to participate.
